code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,693 @@
+	anchored = 0				// Small enough to move even when filled
+	throwpass = 1				// Just like at the county fair, you can't seem to throw the ball in to win the goldfish
+	pass_flags = PASSTABLE		// Small enough to pull onto a table
+
+	tank_type = "bowl"
+	water_capacity = 50			// Not very big, therefore it can't hold much
+	max_fish = 1				// What a lonely fish
+
+	has_lid = 0
+	max_health = 15				// Not very sturdy
+	cur_health = 15
+	shard_count = 0				// No salvageable shards
+
+/obj/machinery/fishtank/tank
+	name = "fish tank"
+	desc = "A large glass tank designed to house aquatic creatures. Contains an integrated water circulation system."
+	icon = 'icons/obj/fish_items.dmi'
+	icon_state = "tank1"
+	density = 1
+	anchored = 1
+	throwpass = 1				// You can throw objects over this, despite it's density, because it's short enough.
+
+	tank_type = "tank"
+	water_capacity = 200		// Decent sized, holds 2 full large beakers worth
+	max_fish = 4				// Room for a few fish
+
+	has_lid = 1
+	max_health = 50				// Average strength, will take a couple hits from a toolbox.
+	cur_health = 50
+	shard_count = 2
+
+
+/obj/machinery/fishtank/wall
+	name = "wall aquarium"
+	desc = "This aquarium is massive! It completely occupies the same space as a wall, and looks very sturdy too!"
+	icon_state = "wall1"
+	density = 1
+	anchored = 1
+	throwpass = 0				// This thing is the size of a wall, you can't throw past it.
+
+	tank_type = "wall"
+	water_capacity = 500		// This thing fills an entire tile,5 large beakers worth
+	max_fish = 10				// Plenty of room for a lot of fish
+
+	has_lid = 1
+	max_health = 100			// This thing is a freaking wall, it can handle abuse.
+	cur_health = 100
+	shard_count = 3
+
+
+
+//////////////////////////////
+//		VERBS & PROCS		//
+//////////////////////////////
+
+/obj/machinery/fishtank/verb/toggle_lid_verb()
+	set name = "Toggle Tank Lid"
+	set category = "Object"
+	set src in view(1)
+	toggle_lid(usr)
+
+/obj/machinery/fishtank/proc/toggle_lid(var/mob/living/user)
+	lid_switch = !lid_switch
+	update_icon()
+
+/obj/machinery/fishtank/verb/toggle_light_verb()
+	set name = "Toggle Tank Light"
+	set category = "Object"
+	set src in view(1)
+	toggle_light(usr)
+
+/obj/machinery/fishtank/proc/toggle_light(var/mob/living/user)
+	light_switch = !light_switch
+	if(light_switch)
+		set_light(2,2,"#a0a080")
+	else
+		set_light(0)
+
+//////////////////////////////
+//		NEW() PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/New()
+	..()
+	fish_list = new/list()
+	egg_list = new/list()
+	if(!has_lid)				//Tank doesn't have a lid/light, remove the verbs for then
+		verbs -= /obj/machinery/fishtank/verb/toggle_lid_verb
+		verbs -= /obj/machinery/fishtank/verb/toggle_light_verb
+
+/obj/machinery/fishtank/tank/New()
+	..()
+	if(prob(5))					//5% chance to get the castle decoration
+		icon_state = "tank2"
+
+//////////////////////////////
+//		ICON PROCS			//
+//////////////////////////////
+
+/obj/machinery/fishtank/update_icon()
```

Comment:
You could even make the overlays new images set to the appearance of the fish inside the tank, and apply a negative scale to the images' transforms to shrink their sprites small enough to fit inside the tank.

That way, the fish you see in the tank are indicative of the fish that are actually inside the tank.